api/client/src/main/java/org/projectnessie/client/NessieClientBuilder.java

```diff
@@ -171,6 +172,16 @@ default Set<String> names() {
   @CanIgnoreReturnValue
   NessieClientBuilder withSSLParameters(SSLParameters sslParameters);
 
+  /**
+   * Registers a callback to cancel an ongoing, blocking client setup.
+   *
+   * <p>When using "blocking" authentication, for example OAuth2 device or code flows, is being
+   * used, users may want to cancel an ongoing authentication. An application can register a
```

Comment:
```suggestion
   * <p>When using "blocking" authentication, for example OAuth2 device or code flows, users
   * may want to cancel an ongoing authentication. An application can register a
```

Also the description is slightly off: there is no "blocking client setup": the OAuth2 authenticator does not block when started (but it may block the first HTTP request, until the first token fetch succeeds).

Reply:
It blocks the first requests - so it is blocking

Reply:
Sure, but client setup is never blocking.